### ISSUE TITLE
DirectWrite integration

### ISSUE THREAD

**Issue by MSleepyPanda** [MSleepyPanda/rust-font-loader#2]
@vvuk wrote Rust bindings for [DirectWrite](https://github.com/vvuk/dwrote-rs) which some applications use internally, e.g. WR/Servo.
Currently, we're using wingdi for font enumeration, but DirectWrite has an interoperabillity interface for wingdi, but this part isn't ported yet.

ToDo:
- [ ] Enhance the gdi interop implementation in DirectWrite
- [ ] Adapt interface

**Comment by vvuk** [comment]
Do you actually need the GDI interop? If dwrite is available, it seems like
the crate should fully use it and not go to GDI at all.

**Comment by MSleepyPanda** [comment]
@vvuk Yes GDI interop is desired, as of the current state of winapi-rs bindings, one can't figure out whether or not a font is monospaced. For reference [MSDN Docs](https://msdn.microsoft.com/de-de/library/windows/desktop/hh780408(v=vs.85).aspx) and [winapi-rs Docs](https://retep998.github.io/doc/winapi/dwrite/struct.IDWriteFontFace.html).
Also, font loading from disk isn't implemented yet, so this would be the most easy way for us to maintain our interface.

I have the corresponding PR almost ready, expect it to come soon :)

**Comment by atouchet** [comment]
For reference the upstream dwrote repo is now located at: https://github.com/servo/dwrote-rs
